packages/rsocket-tck/src/Deferred.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @providesModule Deferred
+ * @typechecks
+ * @flow
+ */
+
+/**
+ * Deferred provides a Promise-like API that exposes methods to resolve and
+ * reject the Promise. It is most useful when converting non-Promise code to use
+ * Promises.
+ *
+ * If you want to export the Promise without exposing access to the resolve and
+ * reject methods, you should export `getPromise` which returns a Promise with
+ * the same semantics excluding those methods.
+ */
+export class Deferred<Tvalue, Treason> {
```

Comment:
Was wondering where we draw the line on what should be copied over to the project from `fbjs` and what should have alternatives provided. For example, in other recent PR's we removed `Deferred` and provided simple alternatives, however here we've moved `Deferred` into the project. What was the motivation for doing so? Was `Deferred` used more heavily in the TCK than in other scenarios?

Reply:
Yes, it's a bit hard to migrate the tck module as it makes heavier use of this class than other modules. But at the same time, the story of the tck module is not clear - I tried to describe the issue in the PR where it's very unclear what it can run against.